diff --git a/src/URI/ByteString/Internal.hs b/src/URI/ByteString/Internal.hs
index 50a7e6b..0daa97c 100644
--- a/src/URI/ByteString/Internal.hs
+++ b/src/URI/ByteString/Internal.hs
@@ -190,7 +190,7 @@ normalizeURIRef' o = BB.toByteString . normalizeURIRef o
 -------------------------------------------------------------------------------
 normalizeURI :: URINormalizationOptions -> URIRef Absolute -> Builder
 normalizeURI o@URINormalizationOptions {..} URI {..} =
-  scheme <> BB.fromString ":" <> normalizeRelativeRef o (Just uriScheme) rr
+  scheme <> BB.fromString "://" <> normalizeRelativeRef o (Just uriScheme) rr
   where
     scheme = bs (sCase (schemeBS uriScheme))
     sCase
@@ -310,7 +310,7 @@ serializeFragment' = BB.toByteString . serializeFragment
 
 -------------------------------------------------------------------------------
 serializeAuthority :: URINormalizationOptions -> Maybe Scheme -> Authority -> Builder
-serializeAuthority URINormalizationOptions {..} mScheme Authority {..} = BB.fromString "//" <> userinfo <> bs host <> port
+serializeAuthority URINormalizationOptions {..} mScheme Authority {..} = userinfo <> bs host <> port
   where
     userinfo = maybe mempty serializeUserInfo authorityUserInfo
     host = hCase (hostBS authorityHost)
