diff --git a/src/URI/ByteString/Internal.hs b/src/URI/ByteString/Internal.hs
index 50a7e6b..1814b36 100644
--- a/src/URI/ByteString/Internal.hs
+++ b/src/URI/ByteString/Internal.hs
@@ -303,7 +303,7 @@ serializeQuery' opts = BB.toByteString . serializeQuery opts
 
 -------------------------------------------------------------------------------
 serializeFragment :: Maybe ByteString -> Builder
-serializeFragment = maybe mempty (\s -> c8 '#' <> urlEncodeQuery s)
+serializeFragment = maybe mempty (\s -> c8 '#' <> bs s)
 
 serializeFragment' :: Maybe ByteString -> ByteString
 serializeFragment' = BB.toByteString . serializeFragment
