diff --git a/src/URI/ByteString/Internal.hs b/src/URI/ByteString/Internal.hs
index 50a7e6b..07153e7 100644
--- a/src/URI/ByteString/Internal.hs
+++ b/src/URI/ByteString/Internal.hs
@@ -320,10 +320,10 @@ serializeAuthority URINormalizationOptions {..} mScheme Authority {..} = BB.from
     port = maybe mempty packPort effectivePort
     effectivePort = do
       p <- authorityPort
-      dropPort mScheme p
+      scheme <- mScheme
+      dropPort scheme p
     packPort (Port p) = c8 ':' <> BB.fromString (show p)
-    dropPort Nothing = Just
-    dropPort (Just scheme)
+    dropPort scheme
       | unoDropDefPort = dropPort' scheme
       | otherwise = Just
     dropPort' s p
