diff --git a/src/URI/ByteString/Internal.hs b/src/URI/ByteString/Internal.hs
index 50a7e6b..1454fea 100644
--- a/src/URI/ByteString/Internal.hs
+++ b/src/URI/ByteString/Internal.hs
@@ -640,8 +640,7 @@ queryParser opts = do
       | otherwise -> fail' MalformedPath
     _ -> pure mempty
   where
-    itemsParser = Query . filter neQuery <$> A.sepBy' (queryItemParser opts) (word8' ampersand)
-    neQuery (k, _) = not (BS.null k)
+    itemsParser = Query <$> A.sepBy' (queryItemParser opts) (word8' ampersand)
 
 -------------------------------------------------------------------------------
 
