diff --git a/src/URI/ByteString/Internal.hs b/src/URI/ByteString/Internal.hs
index 50a7e6b..57f8105 100644
--- a/src/URI/ByteString/Internal.hs
+++ b/src/URI/ByteString/Internal.hs
@@ -335,7 +335,7 @@ serializeAuthority' opts mScheme = BB.toByteString . serializeAuthority opts mSc
 
 -------------------------------------------------------------------------------
 serializeUserInfo :: UserInfo -> Builder
-serializeUserInfo UserInfo {..} = bs uiUsername <> c8 ':' <> bs uiPassword <> c8 '@'
+serializeUserInfo UserInfo {..} = bs uiUsername <> c8 ':' <> bs uiPassword
 
 serializeUserInfo' :: UserInfo -> ByteString
 serializeUserInfo' = BB.toByteString . serializeUserInfo
